### PR TITLE
Config specific.avro.reader to true for Specific Avro SerDe

### PR DESCRIPTION
**Pull request by Ishiihara**

### CODE REVIEW

**Comment by Ishiihara** [comment]
@guozhangwang

File: kafka-streams/src/main/java/io/confluent/examples/streams/utils/SpecificAvroDeserializer.java [hunk 1]
```diff
@@ -41,7 +43,9 @@ public SpecificAvroDeserializer(SchemaRegistryClient client, Map<String, ?> prop
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void configure(Map<String, ?> configs, boolean isKey) {
+        ((Map<String, Object>) configs).put(SPECIFIC_AVRO_READER_CONFIG, true);
```
Comment (guozhangwang) [inline kafka-streams/src/main/java/io/confluent/examples/streams/utils/SpecificAvroDeserializer.java]: Can we just use AbstractKafkaAvroConfig.SPECIFIC_AVRO_READER_CONFIG by importing that class? Ditto below.

Comment (Ishiihara) [inline kafka-streams/src/main/java/io/confluent/examples/streams/utils/SpecificAvroDeserializer.java]: Sure.

**Comment by Ishiihara** [comment]
@guozhangwang PTAL.

**Comment by guozhangwang** [comment]
@Ishiihara Could you file a PR for CP3.0-0.10.0 branch as well?

**Comment by miguno** [comment]
I think this pull request should have updated the src/main/ and src/test/ demos that work on Specific Avro data, too.  For example, my understanding is that, with this PR merged, we don't need to explicitly set this config in the StreamsConfig setup, like here: https://github.com/confluentinc/examples/blob/master/kafka-streams/src/test/java/io/confluent/examples/streams/SpecificAvroIntegrationTest.java#L80.

```java
    Properties producerConfig = new Properties();
    producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
    producerConfig.put(ProducerConfig.ACKS_CONFIG, "all");
    producerConfig.put(ProducerConfig.RETRIES_CONFIG, 0);
    producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
    producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
    producerConfig.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, CLUSTER.schemaRegistryUrl());
    IntegrationTestUtils.produceValuesSynchronously(inputTopic, inputValues, producerConfig);
```

File: kafka-streams/src/main/java/io/confluent/examples/streams/utils/SpecificAvroSerializer.java [hunk 2]
```diff
@@ -41,7 +44,9 @@ public SpecificAvroSerializer(SchemaRegistryClient client, Map<String, ?> props)
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void configure(Map<String, ?> configs, boolean isKey) {
+        ((Map<String, Object>) configs).put(SPECIFIC_AVRO_READER_CONFIG, true);
```
Comment (miguno) [inline kafka-streams/src/main/java/io/confluent/examples/streams/utils/SpecificAvroSerializer.java]: Question: Why is this also required for the serializer?  My understanding was that this setting has an effect only for the deserializer?

**Comment by miguno** [comment]
Note: I updated `master` to remove `KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG` from all examples.  The only exception is `SpecificAvroIntegrationTest` where we need to set this config when configuring the standard Java consumer.

**Comment by miguno** [comment]
Lastly, I also merged this PR and my follow-up change back into `kafka-0.10.0.0-cp-3.0.0`.

**Pull request merged**